### PR TITLE
Training SLAP using stretch data: V1.0

### PR DESCRIPTION
**Pull request by zephirefaith**
## Motivation and Context

- Use data collected on Stretch to train a manipulation policy
- Will achieve: 
  - End-to-end testing of SLAP from perception to output to execution
  - First demo of SLAP on stretch
- This will add/update:
	- [ ] Stretch H5 compatible dataloaders
		- [ ] Use Stretch embodiment for labeling interaction points
		- [ ] Updating keys on h5 collection side + dataloader end to have a consistent schema
	- [ ] Fixes needed for end-to-end training, inference and execution

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: src/home_robot/home_robot/motion/stretch.py
Comment (exhaustin) [inline src/home_robot/home_robot/motion/stretch.py:619]: Should fix or make an issue before merging.

File: src/home_robot/home_robot/motion/stretch.py
Comment (exhaustin) [inline src/home_robot/home_robot/motion/stretch.py:659]: Seems like it's fixed? Should remove this comment.

File: src/home_robot/home_robot/motion/stretch.py
Comment (exhaustin) [inline src/home_robot/home_robot/motion/stretch.py:663]: Add more context on what errors are being raised? Can also be put into an issue.

File: src/home_robot/home_robot/motion/stretch.py
Comment (exhaustin) [inline src/home_robot/home_robot/motion/stretch.py:696]: Add more comments as to what exactly the two `compute_ik` returns? Would make this part easier to change in the future.

File: src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py
Comment (exhaustin) [inline src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py:437]: Can we just remove the `rotate_images` argument?

File: tests/home_robot/motion/test_ik.py
Comment (exhaustin) [inline tests/home_robot/motion/test_ik.py:46]: Why this change?

**Review by exhaustin — CHANGES_REQUESTED** [review]
Looks good in general, mostly had some clarification questions and ask for more comments.

File: src/home_robot/home_robot/motion/stretch.py
Comment (exhaustin) [reply src/home_robot/home_robot/motion/stretch.py:696]: Actually maybe we should wait for #121 to be merged, seems like it deals with a lot of issues here.

File: src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py:437]: we need to verify that it's correct by default for this use case; hopefully we can remove it. does get images even need the flag still?

File: projects/slap_manipulation/src/slap_manipulation/dataloaders/robot_loader.py
Comment (cpaxton) [inline projects/slap_manipulation/src/slap_manipulation/dataloaders/robot_loader.py:597]: dead code - either add a debug code path or just remove it

File: projects/slap_manipulation/src/slap_manipulation/dataloaders/robot_loader.py
Comment (cpaxton) [inline projects/slap_manipulation/src/slap_manipulation/dataloaders/robot_loader.py:619]: maybe we should split this out into a helper function

File: projects/slap_manipulation/src/slap_manipulation/policy/interaction_prediction_module.py
Comment (cpaxton) [inline projects/slap_manipulation/src/slap_manipulation/policy/interaction_prediction_module.py:460]: this will still work with the RLBench and Franka data right?

File: src/home_robot/home_robot/motion/stretch.py
Comment (cpaxton) [inline src/home_robot/home_robot/motion/stretch.py:610]: don't we already have a manip ik pose function that should do this?
if it's pin specific it should probably be in that IK solver?

**Comment by cpaxton** [comment]
overall seems good, some minor stuff

File: projects/slap_manipulation/src/slap_manipulation/policy/interaction_prediction_module.py
Comment (zephirefaith) [reply projects/slap_manipulation/src/slap_manipulation/policy/interaction_prediction_module.py:460]: It's a new function which doesn't disrupt how RLBench + Franka use IPM/APM. I am planning on porting over all code to use this in subsequent PRs.

**Comment by zephirefaith** [comment]
TODOS for self:
- [ ] Confirm the new code works on robot
- [ ] Run hw_test
- [ ] Fix specific functions in `stretch.py` since a consistent IK interface was fixed in #121

File: actions.yaml
Comment (cpaxton) [inline actions.yaml:2]: What is this file for?

File: projects/slap_manipulation/scripts/conf/slap_inference.yaml
Comment (cpaxton) [inline projects/slap_manipulation/scripts/conf/slap_inference.yaml:7]: don't commit local paths like this

File: projects/instanceimagenav/habitat_extensions.py
Comment (cpaxton) [inline projects/instanceimagenav/habitat_extensions.py:99]: Why are you making changes to these files?

File: projects/slap_manipulation/scripts/execution.py
Comment (cpaxton) [inline projects/slap_manipulation/scripts/execution.py:11]: this path will be invalid - should be a relative path like projects/slap_manipulation/actions.yaml

File: projects/slap_manipulation/scripts/execution.py
Comment (cpaxton) [inline projects/slap_manipulation/scripts/execution.py:9]: should probably be def main() with some command line arguments - and should have a comment for how we are expected to use this script?

File: src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py:439]: has this been tested on the robot? can Vidhi run the habitat ovmm code, or can you, before we merge?

File: src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py:157]: why did you make this change?

File: src/home_robot_hw/home_robot_hw/env/stretch_manipulation_env.py
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/stretch_manipulation_env.py:56]: please verify things are still working on the robot

File: src/home_robot_hw/home_robot_hw/remote/modules/manip.py
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/remote/modules/manip.py:201]: why print this?

**Review by cpaxton — CHANGES_REQUESTED** [review]
Some documentation fixes/paths need fixed; needs real-robot tests since it touches home_robot_hw

**Comment by zephirefaith** [comment]
@cpaxton Reverted all changes to non-slap src files. Verified OVMM works with this branch. If tests pass, let's merge?

**Review by cpaxton — APPROVED** [review]
lgtm

**Review by exhaustin — APPROVED** [review]

**Pull request merged**
